### ISSUE TITLE
Enum member named `name` causes due to instance variable of same name

### ISSUE THREAD

**Issue by bijij** [microsoft/pyright#2437]
**Describe the bug**
An enum member named `name` will cause issues with type-checking.

**To Reproduce**
Consider the code described below:

Pyright will yield the following output:
```
test.py:4:12 - error: Expression of type "Literal[1]" cannot be assigned to declared type "str"
    "Literal[1]" is incompatible with "str" (reportGeneralTypeIssues)
test.py:7:14 - error: Cannot access member "value" for type "str"
    Member "value" is unknown (reportGeneralTypeIssues
```

**Expected behavior**
`Foo.name` would be interpreted as a Literal enum member rather than `str`

**Screenshots or Code**
```python
from enum import Enum

class Foo(Enum):
    name = 1


1 + Foo.name.value
```

![image](https://user-images.githubusercontent.com/8677174/137431780-e4acadc3-c65e-4a6b-8c34-85b2d32759d0.png)

**VS Code extension or command-line**
pyright 1.1.178

**Comment by erictraut** [comment]
Yes, the attributes `name` and `value` are reserved by the Enum class and shouldn't be used for enumerated values. Refer to `enum.pyi` for details.

**Comment by bijij** [comment]
> Refer to `enum.pyi` for details.

I figured you were referring to `enum.pyi` in typeshed? I can't see a comment declaring this though in there?

Can you provide a link? Apologies if I'm just blind.
